### PR TITLE
Clean up write_boot_loader

### PR DESCRIPTION
**Pull request by poncovka**
The method `set_boot_args` in the `Bootloader` class now depends only on
the storage object. The method was split to several smaller methods.

The function `write_boot_loader` was split to several smaller functions
and reorganized, so we can later drop the dependency on the payload
object.

### CODE REVIEW

File: pyanaconda/bootloader/base.py [hunk 1]
```diff
@@ -826,20 +800,27 @@ def set_boot_args(self, *args, **kwargs):
...
-                setup_string = cfg_obj.dracutSetupString()
-                self.boot_args.add(setup_string)
-                self.dracut_args.add(setup_string)
+    def _set_graphical_boot_args(self):
+        """Set up the graphical boot."""
+        args = []
+
+        try:
+            import rpm
+        except ImportError:
```
Comment (M4rtinK) [inline pyanaconda/bootloader/base.py]: I wonder if logging a warning there might make sense ? Not sure how serious the situation is though & if it warrants a warning.

Comment (poncovka) [reply pyanaconda/bootloader/base.py]: Fixed.

**Comment by poncovka** [comment]
Updated.

**Comment by poncovka** [comment]
These changes are merged in https://github.com/rhinstaller/anaconda/pull/1804. I messed up the rebase, sorry.

**Comment by jkonecny12** [comment]
That sometimes happened, it's fine :)

**Pull request closed without merging**